src/code/UNOClasses/Deck.java

```diff
@@ -55,4 +56,34 @@ public void shuffleDeck() {
     public int deckTotal() {
         return deck.size();
     }
-}
+
+    public UNOCard deal() {
+        return deck.remove(0);
+    }
+
+    public String toString() {
+        return deck.stream().map(b -> b.toString()).collect(Collectors.joining("\n"));
+    }
+
+    public boolean isEmpty() {
+        return this.deck.isEmpty();
+    }
+
+    public void shuffleDiscardPile(Vector<UNOCard> cards) {
+        this.deck.addAll(cards);
+        shuffleDeck();
+    }
+
+    public static void main(String[] args) {
```

Comment:
You can put this in the main.java file if you don't want to worry about having two mains in our project.

Reply:
I was just doing that to visualize, I don't think it should stay in there.